packages/gatsby/src/utils/babel-loader-helpers.ts

```diff
@@ -136,24 +154,27 @@ const prepareOptions = (
   )
 
   // Go through babel state and create config items for presets/plugins from.
-  const reduxPlugins = []
-  const reduxPresets = []
-  pluginBabelConfig.stages[stage].plugins.forEach(plugin => {
-    reduxPlugins.push(
-      babel.createConfigItem([resolve(plugin.name), plugin.options], {
-        name: plugin.name,
-        type: `plugin`,
-      })
-    )
-  })
-  pluginBabelConfig.stages[stage].presets.forEach(preset => {
-    reduxPresets.push(
-      babel.createConfigItem([resolve(preset.name), preset.options], {
-        name: preset.name,
-        type: `preset`,
-      })
-    )
-  })
+  const reduxPlugins: Array<PluginItem> = []
+  const reduxPresets: Array<PluginItem> = []
+
+  if (stage) {
+    pluginBabelConfig.stages[stage].plugins.forEach(plugin => {
+      reduxPlugins.push(
+        babel.createConfigItem([resolve(plugin.name), plugin.options], {
+          dirname: plugin.name,
```

Comment:
I think adding the `dirname` is ok here, but should we keep the `name` key as well?

https://github.com/babel/babel/blob/main/packages/babel-core/src/config/item.ts

Reply:
I was going to leave it but the types from babel changed and no longer include it.
The `createConfigItem` method that we use here comes from this type https://github.com/babel/babel/blob/main/packages/babel-core/src/config/item.ts#L28 and it now accepts only `dirname` and `type`.

Reply:
Looking at the current babel docs and previous versions the createConfigItem now has these parameters so changing them feels correct